scripts/r.import/r.import.py

```diff
@@ -206,6 +217,25 @@ def main():
                 _("Input <%s> successfully imported without reprojection")
                 % GDALdatasource
             )
+            if flags["e"]:
+                try:
+                    gs.message(_("Calculating estimted resolution..."))
+                    raster_info = gs.raster_info(output)
+                    if raster_info:
+                        estres = (raster_info["ewres"] + raster_info["nsres"])/2.0
+                        gs.message(
+                            _("Estimated target resolution for input band <{out}>: {res}").format(
+                                out=output, res=estres
+                            )
+                        )
+                    else:
+                        gs.warning(
+                            _("Unable to estimate the resolution for the input raster")
+                        )
+                    return 0
+                except CalledModuleError:
+                    gs.fatal(_("Unable to import GDAL dataset <%s>") % GDALdatasource)
+
```

Comment:
No need for `try except` and move this `if flags["e"]` block with `return 0` below the original `try except`
```python
try:
  gs.run_command("r.in.gdal", **parameters)
  ...
  return 0 # <-- delete this line
except
  ...

if flags["e"]:
  ...
return 0
````